include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -185,26 +185,50 @@ using __kernel_name_generator =
     _BaseName<_CustomName>;
 #endif
 
-template <typename _DerivedKernelName>
+template <typename... _KernelNames>
 class __kernel_compiler
 {
+    static constexpr int __n_kernel = sizeof...(_KernelNames);
```

Comment:
I guess `int` is not what we want. `sizeof` should return `std::size_t`. Furthermore, index_sequence should also require `std::size_t`. If it's not so, we should fix that.